### ISSUE TITLE
Confusion with quotation marks in {% wagtailcache %} tag Documentation leads to caching issues

### ISSUE THREAD

**Issue by felix-m-p** [wagtail/wagtail#12541]
### Pertinent section of the Wagtail docs

https://docs.wagtail.org/en/v6.2/topics/writing_templates.html#page-aware-caching

### Details

Hi,

I encountered an issue with the **{% wagtailpagecache %}**  and **{% wagtailcache %}** template tag that may be due to a misunderstanding or oversight in the documentation.

**Issue with the documentation:**

Use the {% wagtailpagecache %} tag in a template with double quotation marks around the cache name, as shown in the documentation:

```html
{% load wagtail_cache %}

{% wagtailpagecache 500 "hero" %}
    <!-- hero -->
{% endwagtailpagecache %}
``` 

Attempt to clear the cache manually in Python:

```python
from django.core.cache import cache
from wagtail.coreutils import make_wagtail_template_fragment_key

key = make_wagtail_template_fragment_key("hero", page, site)
cache.delete(key)  # invalidates cached template fragment
``` 
Which won't delete the cache.

Somehow the invalidation if the Page was edited in the Admin Panel works. As stated in the Django Documentation about Caching, the cache name shouldn't have double quotation marks.
https://docs.djangoproject.com/en/5.1/topics/cache/#template-fragment-caching

**Solution:**
So maybe the Documentation should be changed to that:

```html
 {% load wagtail_cache %}

{% wagtailpagecache 500 hero %}
    <!-- hero -->
{% endwagtailpagecache %}
```

Wagtail version: 6.3
Django version: 5.1.2
Cache backend: Redis

**Comment by ababic** [comment]
@laymonage My two cents: We should probably account for both somehow, as, regardless of the tag, it's common for developers to provide arbitrary strings to tags with quotes. I suspect the best thing to do is add some stripping to the value when assembling the cache key

**Comment by laymonage** [comment]
@ababic Thanks for the input! If using quotes is not the currently expected behaviour, I'm not sure accounting for it is a good idea. It would diverge from Django's `cache` tag behaviour, and depending on how we handle it, it can result in unpredictable behaviour.

For example, if people genuinely want to use quotes in their fragment name, e.g. in the format of `"my view"."my header"."my hero"`, stripping the leading and trailing quotes could result in something like `my view"."my header"."my hero`.

Regardless of that, if we want to allow not using quotes (as per the current actual expected behaviour), then I think our documentation needs to reflect that.

**Comment by laymonage** [comment]
Fixed in #12568. Thanks all!
